UnitTests/Filters/StringFiltersTests.swift

```diff
@@ -12,13 +12,26 @@ class StringFiltersTests: XCTestCase {
   func testLowerFirstWord() {
     let expectations = [
       "string": "string",
-      "string_with_words": "string_with_words",
+      "String": "string",
+      "strIng": "strIng",
+      "strING": "strING",
+      "X": "x",
+      "x": "x",
       "SomeCapString": "someCapString",
+      "someCapString": "someCapString",
+      "string_with_words": "string_with_words",
+      "String_with_words": "string_with_words",
+      "String_With_Words": "string_With_Words",
+      "STRing_with_words": "stRing_with_words",
+      "string_wiTH_WOrds": "string_wiTH_WOrds",
       "": "",
-      "X": "x",
       "URLChooser": "urlChooser",
+      "a__b__c": "a__b__c",
       "__y_z!": "__y_z!",
-      "PLEASESTOPSCREAMING": "pleasestopscreaming"
+      "PLEASESTOPSCREAMING": "pleasestopscreaming",
+      "PLEASESTOPSCREAMING!": "pleasestopscreaminG!",
```

Comment:
From the comments in the code:
- If the string starts with only one uppercase letter, lowercase that first letter
- If the string starts with multiple uppercase letters, lowercase those first letters up to the one before the last uppercase one

e.g. "PeoplePicker" gives "peoplePicker" but "URLChooser" gives "urlChooser"

We should modify the filter to ignore non-letters for testing.